evalml/preprocessing/data_splitters/balanced_classification_sampler.py

```diff
@@ -77,6 +76,7 @@ def fit_resample(self, X, y):
         Returns:
             list: Indices to keep for training data
         """
+        random_state = np.random.RandomState(self.random_seed)
```

Comment:
The fix itself is just to recreate the random number generator used by the undersampling on each invocation.

Reply:
Will this need to propagate to #2079 ?

Reply:
I don't think the oversamplers suffer from this problem but it might be good to add coverage